src/components/Modal.tsx

```diff
@@ -48,13 +49,24 @@ export const Modal = (props: {
 const useBodyRoot = () => {
   const [div, setDiv] = useState<HTMLDivElement | null>(null);
 
+  const isMobile = useIsMobile();
+  const isMobileRef = useRef(isMobile);
+  isMobileRef.current = isMobile;
+
+  useLayoutEffect(() => {
+    if (div) {
+      div.classList.toggle("excalidraw--mobile", isMobile);
+    }
+  }, [div, isMobile]);
+
   useLayoutEffect(() => {
     const isDarkTheme = !!document
       .querySelector(".excalidraw")
       ?.classList.contains("theme--dark");
     const div = document.createElement("div");
 
     div.classList.add("excalidraw", "excalidraw-modal-container");
+    div.classList.toggle("excalidraw-modal-container", isMobileRef.current);
```

Comment:
Doesn’t this duplicate the other layout effect?

Reply:
The other is updating on `isMobile`, whereas this one is mount-only.

Reply:
but it should toggle `excalidraw--mobile`. A typo.

Reply:
This fixes the Modal positioning you were seeing.